test/src/fvkernels/FVKTLimitedAdvection.C

```diff
@@ -0,0 +1,58 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "FVKTLimitedAdvection.h"
+#include "Limiter.h"
+
+registerMooseObject("MooseTestApp", FVKTLimitedAdvection);
+
+using namespace Moose::FV;
+
+InputParameters
+FVKTLimitedAdvection::validParams()
+{
+  InputParameters params = FVFluxKernel::validParams();
+  params.addClassDescription(
+      "Residual contribution from advection operator for finite volume method.");
+  params.addRequiredParam<RealVectorValue>("velocity", "Constant advection velocity");
+  params.addRequiredParam<MaterialPropertyName>(
+      "max_abs_eig", "The maximum absolute value of the advection Jacobian eigenvalues.");
+  params.addParam<MooseEnum>(
+      "limiter", moose_limiter_type, "The limiter to use for the advected quantity.");
+  params.addParam<bool>("add_artificial_diff", true, "Add artificial diffusion for monotonicity.");
+  return params;
+}
+
+FVKTLimitedAdvection::FVKTLimitedAdvection(const InputParameters & params)
+  : FVFluxKernel(params),
+    _velocity(getParam<RealVectorValue>("velocity")),
+    _limiter(Limiter::build(LimiterType(int(getParam<MooseEnum>("limiter"))))),
+    _cd_limiter(Limiter::build(LimiterType::CentralDifference)),
+    _grad_u_elem(_var.adGradSln()),
+    _grad_u_neighbor(_var.adGradSlnNeighbor()),
+    _max_abs_eig(getADMaterialProperty<Real>("max_abs_eig")),
+    _add_artificial_diff(getParam<bool>("add_artificial_diff"))
+{
+}
+
+ADReal
+FVKTLimitedAdvection::computeQpResidual()
```

Comment:
Why these kernels are in test folder? Are not general enough? It seems very  general for me

Reply:
I think that in practice users are going to be solving problems in which the velocity is provided via variables. Moreover, determination of `max_abs_eig` will be a function of the velocity and the function will likely be defined in the kernel implementing the KT method, e.g. in `PCNSFVKT`, as opposed to provided through an arbitrary material property or simple user input. @aeslaughter was recently lamenting the explosion of objects in the framework. I think that until somebody asks for this to be available, I think this is best in test because I don't think it's generally useful.

Reply:
Got it.  Class name seem pretty general though, but we might not be worried about